src/views/home/components/transaction-form/transaction-form.view.tsx

```diff
@@ -100,6 +107,11 @@ const TransactionForm: FC<TransactionFormProps> = ({ onSubmit, transaction, acco
     }
   }, [chains, account]);
 
+  useEffect(() => {
+    onChainChange();
+    setAmount(undefined);
+  }, [onChainChange]);
```

Comment:
In this case if you've `transaction` you don't see changes if Metamask changes network.
And we reset `amount` because we do the same when user change chains clicking buttons form